CMakeLists.txt

```diff
@@ -77,95 +61,145 @@ elseif(${CMAKE_BUILD_TARGET} MATCHES "Both")
+    PRIVATE
+      LibUVC::UVC
+      Threads::Threads
+  )
 endif()
 
 if(BUILD_TEST)
-
-    # Try to find OpenCV using a module or pkg-config. If that doesn't work, search for the header.
-  find_package(opencv QUIET)
-  if(OPENCV_FOUND)
-    set(OPENCV_LINK_FLAGS ${OPENCV_LIBRARIES})
-  else()
-    pkg_check_modules(OPENCV QUIET opencv)
-    if(OPENCV_FOUND)
-        set(OPENCV_INCLUDE_DIR ${OPENCV_INCLUDE_DIRS})
-        set(OPENCV_LINK_FLAGS ${OPENCV_LDFLAGS})
-    endif()
-  endif()
+  find_package(OpenCV
+    REQUIRED COMPONENTS
+      opencv_core
+      opencv_highgui
+  )
 
   add_executable(uvc_test src/test.c)
-  target_link_libraries(uvc_test uvc ${OPENCV_LDFLAGS})
+  target_link_libraries(uvc_test
+    PRIVATE
+      LibUVC::UVC
+      opencv_core
+      opencv_highgui
+  )
 endif()
 
-install(TARGETS ${UVC_TARGETS}
-    EXPORT libuvcTargets
-  LIBRARY DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/${CMAKE_LIBRARY_ARCHITECTURE}"
-  ARCHIVE DESTINATION "${CMAKE_INSTALL_PREFIX}/lib/${CMAKE_LIBRARY_ARCHITECTURE}"
-  PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_PREFIX}/include/libuvc"
-)
-
-export(TARGETS ${UVC_TARGETS}
-  FILE "${PROJECT_BINARY_DIR}/libuvcTargets.cmake")
-export(PACKAGE libuvc)
 
-set(CONF_INCLUDE_DIR "${CMAKE_INSTALL_PREFIX}/include")
-set(CONF_LIBRARY_DIR "${CMAKE_INSTALL_PREFIX}/lib/${CMAKE_LIBRARY_ARCHITECTURE}")
-set(CONF_LIBRARY "${CMAKE_INSTALL_PREFIX}/lib/${CMAKE_LIBRARY_ARCHITECTURE}/${CMAKE_SHARED_LIBRARY_PREFIX}uvc${CMAKE_SHARED_LIBRARY_SUFFIX}")
+include(GNUInstallDirs)
+set(CMAKE_INSTALL_CMAKEDIR ${CMAKE_INSTALL_LIBDIR}/cmake/libuvc)
+
+install(
+  TARGETS ${UVC_TARGETS}
+  EXPORT libuvcTargets
+  LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR}
+  RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  INCLUDES DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
+  PUBLIC_HEADER DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}/libuvc
+)
 
-configure_file(libuvcConfig.cmake.in ${PROJECT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/libuvcConfig.cmake)
+install(EXPORT libuvcTargets
+  FILE libuvcTargets.cmake
+  NAMESPACE LibUVC::
+  DESTINATION ${CMAKE_INSTALL_CMAKEDIR}
+)
 
-configure_file(libuvcConfigVersion.cmake.in ${PROJECT_BINARY_DIR}/libuvcConfigVersion.cmake @ONLY)
+install(FILES 
+    cmake/FindLibUSB.cmake
+    cmake/FindJpegPkg.cmake
+  DESTINATION ${CMAKE_INSTALL_CMAKEDIR}
+)
 
-configure_file(libuvc.pc.in ${PROJECT_BINARY_DIR}/libuvc.pc @ONLY)
+include(CMakePackageConfigHelpers)
+write_basic_package_version_file(libuvcConfigVersion.cmake
+  COMPATIBILITY AnyNewerVersion
+)
 
 install(FILES
-  "${PROJECT_BINARY_DIR}${CMAKE_FILES_DIRECTORY}/libuvcConfig.cmake"
-  "${PROJECT_BINARY_DIR}/libuvcConfigVersion.cmake"
-  DESTINATION "${INSTALL_CMAKE_DIR}")
+  ${CMAKE_CURRENT_BINARY_DIR}/libuvcConfigVersion.cmake
+  DESTINATION ${CMAKE_INSTALL_CMAKEDIR}
+)
 
-install(EXPORT libuvcTargets
-  DESTINATION "${INSTALL_CMAKE_DIR}")
+if(JPEG_FOUND)
+  set(PKGCONFIG_JPEG_LDFLAG "-ljpeg")
+endif()
```

Comment:
If we use JPEG library we need to add linker flag for it in config file for pkgconfig